test/prow/testcases/common/dataset_pvc_subpath.py

```diff
@@ -0,0 +1,200 @@
+"""
+TestCase: Dataset with PVC Subpath Verification
+DDC Engine: Alluxio
+Steps:
+1. create PV and PVC
+2. create Dataset(PVC subpath) & Runtime
+3. check if dataset is bound
+4. check if persistentVolumeClaim & PV is created
+5. check if app pod is running and if data read success
+6. clean up
+"""
+
+import os
+import shutil
+import sys
+
+project_root = os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
+sys.path.insert(0, project_root)
+
+import fluid.fluidapi as fluidapi
+import fluid.step_funcs as funcs
+from framework.step import SimpleStep, StatusCheckStep, currying_fn, dummy_back
+from framework.testflow import TestFlow
+from kubernetes import client, config
+from kubernetes.stream import stream
+
+
+def createPVandPVC(pv_name, pvc_name, namespace="default"):
+    # create a local path for PV mounts
+    pvpath = os.path.join(os.getcwd(), 'pvpath')
+    pv_subpath = os.path.join(pvpath, 'subpath')
+    os.makedirs(pv_subpath, exist_ok=True)
```

Comment:
The e2e test cases run in a Pod, so making dirs won't take effect when specifying local pv & pvc. What about taking a Dataset as a the PV's source? For example:

The Dataset for PV is like:
```
kind: Dataset
metadata:
  name: hbase
spec:
  mounts:
    - mountPoint: https://mirrors.bit.edu.cn/apache/zookeeper/stable/
      name: zookeeper
      path: /subpath/zookeeper
```

The PVC subpath dataset is like:
```
kind: Dataset
metadata:
  name: pvc-subpath
spec:
  mounts:
    - mountPoint: pvc://hbase/subpath/
      name: mypv
      path: /
```

So given any app pod that mounts the `pvc-subpath` Dataset's pvc to `/data`, we can check if it works by running `kubectl exec -it <app_pod> -- ls -l /data/zookeeper`.

Reply:
ok, I will take a try.